### PR TITLE
Add minimal, black-compatible flake8 configuration (Kedro 0.17.0)

### PR DESCRIPTION
**Pull request by DmitriiDeriabinQB**
## Description
Applying the changes from #524 to `develop`

## Checklist

- [ ] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md) file
- [ ] Added tests to cover my changes

## Notice

- [ ] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: kedro/templates/project/{{ cookiecutter.repo_name }}/setup.cfg [hunk 1]
```diff
@@ -1,3 +1,7 @@
+[flake8]
+max-line-length=88
+extend-ignore=E203
+
```
Comment (lorenabalan) [inline kedro/templates/project/{{ cookiecutter.repo_name }}/setup.cfg:4]: This should be replicated in `features/steps/test_starter/` as well I believe.

**Review by 921kiyo — APPROVED** [review]
👍

**Comment by lorenabalan** [comment]
Can you also add the changes to the `0.17.0` branches of the starters? 🙏

**Review by lorenabalan — APPROVED** [review]
Approving assuming the tests are fixed.

**Comment by DmitriiDeriabinQB** [comment]
> Can you also add the changes to the `0.17.0` branches of the starters? 🙏

@lorenabalan did you mean something like quantumblacklabs/kedro-starter-spaceflights#6? Or did I miss something else?

**Comment by lorenabalan** [comment]
> @lorenabalan did you mean something like [quantumblacklabs/kedro-starter-spaceflights#6](https://github.com/quantumblacklabs/kedro-starter-spaceflights/pull/6)? Or did I miss something else?

Oh that's exactly what I mean, ignore me!! ⭐ ⭐ ⭐

**Pull request merged**